components/WalletProvider.tsx

```diff
@@ -90,14 +88,26 @@ export const WalletProvider = ({
     [address, disconnect]
   )
 
+  const handleChainChanged = useCallback(
+    ({ chainId }) => {
+      console.log('Chain changed to', chainId)
+      setChainId(chainId)
+    },
+    [setChainId]
```

Comment:
I notice a difference in this dependancy array where it contains a setState (setChainId) rather that the constant (chainId), was this intended?